### PR TITLE
Multiplayer Chat - Clickable Links

### PR DESCRIPTION
**Pull request by Sixmorphugus**
- Strings beginning with "http://" in the chat are now parsed into clickable links that open the web browser on windows (and do nothing on other platforms because the functions are currently stubs)
- When a player's group is changed, a message displays.
- Player names are now yellow in all ui.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
You'll have to add support for https:// links as well.

**Comment by Sixmorphugus** [comment]
I've noticed an issue in multiplayer where changing the group sometimes changes the wrong person's.

**Comment by HaasJona** [comment]
For Linux, this code works:

```c
void platform_open_browser(const char* url) {
	int exit_value;
	char cmd[MAX_PATH];

	snprintf(cmd, MAX_PATH, "%s %s", "xdg-open", url);

	execute_cmd(cmd, &exit_value, 0, 0 );
}
```

**Comment by Sixmorphugus** [comment]
@HaasJona 
Thanks, let me put that in.

**Comment by janisozaur** [comment]
@HaasJona good job, that's exactly what I had hoped for.

**Comment by Gymnasiast** [comment]
I believe `open` is the OS X equivalent of `xdg-open`, but I'd like to see that confirmed by someone who knows more about OS X. @LRFLEW can you say more about this.

**Comment by HaasJona** [comment]
I don't have a mac to test it, but yes, I know from a colleague that "open" is the osx equivalent.
Am 25.03.2016 4:18 nachm. schrieb Michael Steenbeek <notifications@github.com>:I believe open is the OS X equivalent of xdg-open, but I'd like to see that confirmed by someone who knows more about OS X. @LRFLEW can you say more about this.

**Comment by Gymnasiast** [comment]
@HaasJona : OpenRCT2 doesn't even run on 64-bit Ubuntu 14.04 and has several other high version requirements, so catering for systems so old they lack xdg-open seems to a wasted effort to me.

File: src/network/network.cpp [hunk 1]
```diff
@@ -2217,6 +2216,10 @@ void network_set_player_group(unsigned int index, unsigned int groupindex)
 	gNetwork.Server_Send_CHAT(str_sg);
 }
 
+void network_set_player_group(unsigned int playerIndex, unsigned int groupIndex) {
```
Comment (zsilencer) [inline src/network/network.cpp]: this looks like it just does the same thing, what issue does it fix exactly?

**Comment by LinusU** [comment]
@Sixmorphugus Why did you remove OS X `open` (a60ac0b)? It should work perfectly :+1:

**Comment by janisozaur** [comment]
@LinusU `execute_cmd` is only available on Linux.

**Comment by LinusU** [comment]
Ahh, that explains it... `system`?

**Comment by LRFLEW** [comment]
`open` does seem to work from my testing, but it also seems to be the wrong way of doing it effectively. I'm fairly certain there's a way of doing it through the API (which will likely be equivalent, but more verbose and efficient). I'll look into it, and PR the PR if I come up with something.

**Comment by wolfreak100** [comment]
I'd like if this was an option... and maybe a yes/no confirmation window asking "open ___ in browser?", or even just being required to double click it, just to help prevent accidental clicks.

**Comment by PFCKrutonium** [comment]
If I may suggest two things: A linking permission, and a server wide link disable setting.

**Comment by lopezloo** [comment]
How does it work if OpenRCT2 is launched in Steam overlay?

File: data/language/english_uk.txt [hunk 2]
```diff
@@ -4094,6 +4094,7 @@ STR_5786    :Invalid group name
 STR_5787    :{COMMA32} players online
 STR_5788    :Default inspection interval:
 STR_5789    :Disable lightning effect
+STR_5799    :{OUTLINE}{YELLOW}{STRING}{WHITE} was moved to group {YELLOW}{STRING}.
```
Comment (Nubbie) [inline data/language/english_uk.txt]: This string now collides with the merged #3254 & later on with PR #3192 
Should be changed to STR_5804

**Comment by IntelOrca** [comment]
There are some improvements I would like with this:
* Multiple links in a message.
* Hotspot should only be around the link, not the whole message (and empty space)
* Hotspot should change cursor to hand pointer.

**Comment by janisozaur** [comment]
@Sixmorphugus are you still working on this?

**Comment by Sixmorphugus** [comment]
Yes, have just been very busy with some other stuff.
I'll get onto this shortly.

**Comment by Nubbie** [comment]
The question is if someone could keep on working with this?

**Comment by wolfreak99** [comment]
YEAH I WILL LOL I NEED MORE PUL LREQWUETS WITH MOY NAMEA IN ON IT!

**Pull request closed without merging**